tests/periph_cpuid/tests/01-run.py

```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env python3
+
+# Copyright (C) 2019 Inria
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+
+import os
+import sys
+from testrunner import run
+
+
+CPUID_LEN = int(os.getenv('CPUID_LEN') or 4)
+
+
+def testfunc(child):
+    child.expect_exact('Test for the CPUID driver')
+    child.expect_exact('This test is reading out the CPUID of the platforms CPU')
+    expected = 'CPUID:' + CPUID_LEN * r' 0x[0-9a-fA-F]{2}'
```

Comment:
Why not do

```suggestion
    expected = CPUID:( 0x[0-9a-fA-F]{2})+'
```

This way `CPUID_LEN` does not have to be known beforehand. Or just print it before this check in the firmware. The current approach seems not very scalable to me and requires a lot of manual labor when porting new platforms and also leaves room for errors when the actual `CPUID_LEN` is greater than the `CPUID_LEN` given by the environment.